security/pkg/caclient/client.go

```diff
@@ -31,24 +34,33 @@ import (
 // CAClient is a client to provision key and certificate from the upstream CA via CSR protocol.
 type CAClient struct {
 	platformClient         platform.Client
-	protocolClient         grpc.CAGrpcClient
-	istioCAAddress         string
 	maxRetries             int
 	initialRetrialInterval time.Duration
+	grpcClient             pb.IstioCAServiceClient
```

Comment:
If we couple the CAClient with the IstioCAService proto, this may prevent us from reusing the CAClient code to talk to other non-Istio CAs. I think ideally we can make the CAClient general enough so that in the future, it can support talking to various CAs. WDYT?

Reply:
Definitely we need to extend to support other CA. although I feel it's a little bit of tricky to generalize the interface before determining what other CA's protocol (http or grpc, proto field alighnment, etc) will look like.

Since we already have GAGrpcClient interface, how about:
- We rename it to CAConnection (or CAProtocol, or something else)?
- Make grpc as an implementation, stores the grpc connection in the struct to avoid re-dial for every request.
- Add more implemetation for the  implementation when needed.

Reply:
That makes sense to me. Thanks :)